base/util.go

```diff
@@ -1046,15 +1046,15 @@ func InjectJSONProperties(b []byte, kvPairs ...KVPair) (new []byte, err error) {
 		case int, int8, int16, int32, int64:
 			valBytes = []byte(strconv.Itoa(kv.Val.(int)))
 		case uint, uint8, uint16, uint32, uint64:
-			valBytes = []byte(strconv.Itoa(int(kv.Val.(uint))))
+			valBytes = []byte(strconv.FormatUint(v.(uint64), 10))
```

Comment:
I think a type conversion (i.e. uint64(v)) would be more appropriate here than a type assertion.

Reply:
I don't think this is possible because v is of type interface at that point.

Reply:
Ah yeah you're right, but it does mean we're doing two type switches/assertions on kv.Val here - a bit inefficient.  Based on some similar patterns in the standard library, it seems like the idiomatic way of doing this might be to have separate case statements for uint, uint8, uint16, etc.